client/js/userTaskView.js

```diff
@@ -0,0 +1,185 @@
+            userTasks.forEach(function(rowData) {
+                var row = document.createElement('tr');
+
+                columnsToDisplay.forEach(function(key) {
+                    var cell = document.createElement('td');
+                    var textContentSpan = document.createElement('span');
+
+                    if (key === 'col7') {
+                        textContentSpan.textContent = rowData[key];
+                        setTaskStatusStyle(textContentSpan);
+                        cell.appendChild(textContentSpan);
+                    } else {
+                        cell.textContent = rowData[key];
+                    }
+
+                    if (key !== 'col2' && key !== 'col4') {
+                        cell.classList.add('text-center');
+                    }
+
+                    row.appendChild(cell);
+                });
+
+                tbody.appendChild(row);
+            });
+        }
+    }
+
+    function setTaskStatusStyle(span) {
+        if (span.textContent === 'In progress') {
+            span.classList.add('inProgress');
+        } else if (span.textContent === 'ToDo') {
+            span.classList.add('toDo');
+        } else if (span.textContent === 'Done') {
+            span.classList.add('done');
+        } else if (span.textContent === 'Ready for test') {
+            span.classList.add('readyForTest');
+        } else if (span.textContent === 'Blocked') {
+            span.classList.add('blocked');
+        } else if (span.textContent === 'Code review') {
+            span.classList.add('codeReview');
+        }
+    }
+
+    function sortTable(columnIndex) {
+        let isAscending = true;
+
+        let thClicked = table.querySelectorAll('th')[columnIndex];
+        const key = thClicked.getAttribute('data-key');
+
+        table.querySelectorAll('th').forEach(function(th, index) {
+            if (index !== columnIndex) {
+                th.classList.remove('ascending', 'descending');
+            }
+        });
+
+        if (thClicked.classList.contains('ascending')) {
+            isAscending = false;
+        }
+
+        data.sort(function(a, b) {
+            let valueA = a[key].toUpperCase();
+            let valueB = b[key].toUpperCase();
+
+            if (isAscending) {
+                return valueA.localeCompare(valueB);
+            } else {
+                return valueB.localeCompare(valueA);
+            }
+        });
+
+        thClicked.classList.toggle('ascending', isAscending);
+        thClicked.classList.toggle('descending', !isAscending);
+
+        populateTable(data);
+    }
+
+    // Function to filter tasks by selected user
+    function selectUser(selectedUser) {
+        var filteredData = data.filter(function (task) {
+            // Assuming user information is stored in the task data
+            return selectedUser !== null ? (selectedUser === '' ? !task.user : task.user === selectedUser) : true;
+        });
+
```

Comment:
Here we could make it a bit more interactive , so when the user selects an item from the list I feel like the text should also change to reflect that. For example,
Adding, on like line 15:
`const userDropdownButton = document.getElementById('userDropdown');`

and here each time the selectUser function is called:
`if (selectedUser === null || selectedUser === '') {
userDropdownButton.textContent = selectedUser === null ? 'All users' : 'Unassigned';
} else {
userDropdownButton.textContent = selectedUser;
 }
`